csrc/device_lower/analysis/sync_information.cpp

```diff
@@ -499,7 +499,12 @@ SyncMap::SyncMap(Fusion* fusion, bool error_on_failure) {
         if (error_on_failure) {
           if (raw_dims.hasBID()) {
             NVF_ERROR(
-                producer->getMemoryType() == MemoryType::Global,
+                // We need to allow the output of block quantization and the
+                // outputs of the reshape preceding the block
+                // quantization to be in local memory.
+                producer->getMemoryType() == MemoryType::Global ||
```

Comment:
Instead of adding a special case here, try not setting the BID flag in this case